### PR TITLE
Allow using attributes

### PR DESCRIPTION
**Pull request by rdohms**
Rebased version of #19.

> Hi,
> I'm sending a few improvements, including implementation of @lcobucci's failing test case concerning prophets in class property. It now depends on dev version of PHPStan but after you test it and tell me it's okay, I can release the change.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,10 @@
     ],
     "require": {
         "php": "^7.1",
-        "phpstan/phpstan": "^0.10"
+        "phpspec/prophecy": "^1.7",
```
Comment (Jean85) [inline composer.json]: As done elsewhere in similar extensions, this shouldn't require the packages directly. Also, they're already dev-required

Comment (rdohms) [reply composer.json]: 👍

**Comment by rdohms** [comment]
Any progress on merging this?

**Comment by rdohms** [comment]
Yes, but i guess now we need another rebase

**Comment by rdohms** [comment]
@Jan0707 once you merge #37 i'll rebase this, but please let's not let it stale once more.

**Comment by Jan0707** [comment]
@rdohms merged #37 :)

**Comment by rdohms** [comment]
@Jan0707 there you go.

**Pull request merged**